Yank/utils.py

```diff
@@ -1051,9 +1058,46 @@ def get_keyword_args(function):
         function that do not have a default value will not be included.
 
     """
-    argspec = inspect.getargspec(function)
-    kwargs = argspec.args[len(argspec.args) - len(argspec.defaults):]
-    kwargs = {arg: value for arg, value in zip(kwargs, argspec.defaults)}
+
+    master_mro_stack = []
+
+    def stack_up_argspec(input_argspec):
+        defaults = input_argspec.defaults
+        if defaults is None:
+            defaults = []
+        n_defaults = len(defaults)
+        n_args = len(input_argspec.args)
+        # Cycle through the kwargs only
+        cycle_kwargs = input_argspec.args[n_args - n_defaults:]
+        cycle_kwargs = {arg: value for arg, value in zip(cycle_kwargs, defaults)}
+        # Handle the kwonlyargs for calls with `def F(a,b *args, x=True, **kwargs)
+        if input_argspec.kwonlydefaults is not None:
+            cycle_kwargs = {**cycle_kwargs, **input_argspec.kwonlydefaults}
+        return cycle_kwargs
+
+    def build_argspec(input_function, input_cls):
```

Comment:
I'm not sure there's the need for recursion and to keep track of the explored classes as `getmro()` should already transform the recursive part of the problem into loop form.

Maybe there's something I'm missing but, what happens with something like this:
```python
argspec = inspect.getfullargspec(function)
kwargs = stack_up_argspec(argspec)

if mro_cls is not None and argspec.varkw is not None:
    try:
        mro = inspect.getmro(mro_cls)
    except AttributeError:
        # No MRO, input_function is not part of a class
        mro = [mro_cls]

    for cls in mro[1:]:
        try:
            parent_function = getattr(cls, function.__name__)
        except AttributeError:
            # class of the MRO does not have the method
            pass
        else:
            argspec = inspect.getfullargspec(parent_function)
            kwargs = {**stack_up_argspec(argspec), **kwargs}
return kwargs
```

Reply:
Sure, each step of the recursion internally will recursively do the right thing, but its redundant. consider an MRO stack of classes `(X, Y, Z)`.
`X` will queue `Y` and `Z` to search, but `Y` will *also* search `Z`, so `Z` gets searched twice, yields the same result, and the `X` search through `Z` will replace the `Y` search through `Z`. So I made a master MRO list to avoid the extra work.

I see what you mean about the non-recursive search. I'll fix that

Reply:
I remembered why I wanted the recursion, its so that the `argspec.varkw` could be checked for each MRO and terminated as needed. But I also realize that the way I had it was insufficient since they would all be checked anyways, unless all but one class did not have the `varkw` (`**kwargs)`. It might be fixable with a bunch of cross-mro checks inspecting diffs of the mro lists, but probably not worth putting in at this time.